app/code/Magento/CatalogInventory/Api/Data/InventoryInterface.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace Magento\CatalogInventory\Api\Data;
+
+use Magento\CatalogInventory\Api\LocationInformationInterface;
+use Magento\CatalogInventory\Api\ConfigurationInterface;
+
+/**
+ * Inventory interface
+ *
+ * @api
+ */
+interface InventoryInterface extends LocationInformationInterface
```

Comment:
So to summarize the logic:

Entity Description | Interface for Configuring Data in Backend  | Interface for Business Logic
--------|----------------------|---------------------
Scope of product inventory availability | WarehouseInterface | InventoryInterface
Availability of product in particular scope | WarehouseRecordInterface | InventoryRecordInterface / InventoryIndexRecordInterface 
Product Inventory Reservation | No stock reservation is manageable manualy | InventoryReservationManagementInterface creates InventoryReservationInterface instances to process mutations of stock on frontend. Idea that you cannot directly InventoryRecordInterface, only via InventoryReservationInterface.